api/models/workflow.py

```diff
@@ -797,3 +804,204 @@ def from_variable(cls, *, app_id: str, conversation_id: str, variable: Variable)
     def to_variable(self) -> Variable:
         mapping = json.loads(self.data)
         return variable_factory.build_conversation_variable_from_mapping(mapping)
+
+
+# Only `sys.query` and `sys.files` could be modified.
+_EDITABLE_SYSTEM_VARIABLE = frozenset(["query", "files"])
+
+
+def _naive_utc_datetime():
+    return datetime.now(UTC).replace(tzinfo=None)
+
+
+class WorkflowDraftVariable(Base):
+    UNIQUE_INDEX_APP_ID_NODE_ID_NAME = "workflow_draft_variables_app_id_key"
+
+    __tablename__ = "workflow_draft_variables"
+    __table_args__ = (
+        PrimaryKeyConstraint("id", name="workflow_draft_variables_pkey"),
```

Comment:
The `PrimaryKeyConstraint` can be removed because you used `primary_key=True`.

Reply:
Ah, good point! I was mostly following the definition of another model and hadn’t given it much thought.